packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.tsx

```diff
@@ -61,16 +61,19 @@ export const SimpleFormIterator = (props: SimpleFormIteratorProps) => {
     } = props;
     const [confirmIsOpen, setConfirmIsOpen] = useState<boolean>(false);
     const { append, fields, move, remove, replace } = useArrayInput(props);
-    const { resetField } = useFormContext();
+    const { resetField, trigger } = useFormContext();
     const translate = useTranslate();
     const record = useRecordContext(props);
     const initialDefaultValue = useRef({});
 
     const removeField = useCallback(
         (index: number) => {
             remove(index);
+            // Trigger validation on the Array to avoid ghost errors.
+            // Otherwise, validation errors on removed fields might still be displayed
+            trigger(source);
```

Comment:
Triggering validation on remove is unexpected. Can we do that only if we're using scalar values?